htdocs/PI/write/PIWriteRequest.php

```diff
@@ -179,64 +179,78 @@ private function getAndProcessURL($method, $requestUrl) {
-        if(!is_null($requestArray)) {
-            /*The general case expects a single value to be given. In the specific case
-            * of updating extension properties where no specific property is specified
-            * in the url (and so now in $this->entityPropertyKey) we expect multiple
-            * key/value pairs. We use $this->entityPropertyValue and $this->entityPropertyKVArray
-            * frespectivly for these cases.
-            */
-            if($this->entityProperty!='extensionproperties'||!is_null($this->entityPropertyKey)) {
-                if (isset($requestArray['value']) && count($requestArray)==1) {
-                    $this->entityPropertyValue=$requestArray['value'];
-                } elseif(!isset($requestArray['value'])) {
-                  $this->exceptionWithResponseCode(400,
-                    "A value for \"$this->entityProperty\" should be provided. " .
-                    "This should be provided in a JSON string {\"value\":\"<value for " .
-                    "\"$this->entityProperty\">\"}, with no other pairs present." .
-                    " If you believe \"$this->entityProperty\" should take multiple key/value ".
-                    "pairs, check your spelling of \"$this->entityProperty\"" . $genericError
-                    );
-                } else {
-                  $this->exceptionWithResponseCode(400,
-                    "Request message contained more than one object. " . $genericError
-                  );
-                }
-            } else {
-                if(!empty($requestArray)) {
-                    $this->entityPropertyKVArray=$requestArray;
-                } else {
-                  $this->exceptionWithResponseCode(400,
-                    "Please specify the properties you wish to change. " . $genericError
-                  );
-                }
+    switch ($this->entityProperty) {
+      case 'extensionproperties':
 
-            }
-        } elseif (!empty($requestContents)) {
+        //If a property key has been specified, then we don't expect a K/V list
+        //and want to make use of the default case
+        if(is_null($this->entityPropertyKey)) {
+          $this->entityPropertyKVArray=$requestArray;
+          break;
+        }
+        
+      default:
+
+        //If there is not a value in the array at this point throw an exception
+        if (!isset($requestArray['value'])) {
           $this->exceptionWithResponseCode(400,
-            "The JSON message is not correctly formatted. " . $genericError
+            "A value for \"$this->entityProperty\" should be provided. " .
+            "This should be provided in a JSON string {\"value\":\"<value for " .
+            "\"$this->entityProperty\">\"}, with no other pairs present." . $genericError
           );
         }
-    }
+
+        //If there are additional entiries in our array to the value, throw exception
+        if(count($requestArray)==1) {
```

Comment:
Seems this should not be `if(count($requestArray)==1)`, greater than 1? not equal to one?

Reply:
yes

Reply:
resolved in latest commit